x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/softclassification/AucRoc.java

```diff
@@ -145,16 +146,23 @@ private String restLabelsAggName(ClassInfo classInfo) {
     public EvaluationMetricResult evaluate(ClassInfo classInfo, Aggregations aggs) {
         Filter classAgg = aggs.get(evaluatedLabelAggName(classInfo));
         Filter restAgg = aggs.get(restLabelsAggName(classInfo));
-        double[] tpPercentiles = percentilesArray(classAgg.getAggregations().get(PERCENTILES));
-        double[] fpPercentiles = percentilesArray(restAgg.getAggregations().get(PERCENTILES));
+        double[] tpPercentiles = percentilesArray(classAgg.getAggregations().get(PERCENTILES),
+            "[" + getMetricName() + "] requires at least one actual_field to have the value [" + classInfo.getName() + "]");
+        double[] fpPercentiles = percentilesArray(restAgg.getAggregations().get(PERCENTILES),
+            "[" + getMetricName() + "] requires at least one actual_field not to have the value [" + classInfo.getName() + "]");
         List<AucRocPoint> aucRocCurve = buildAucRocCurve(tpPercentiles, fpPercentiles);
         double aucRocScore = calculateAucScore(aucRocCurve);
         return new Result(aucRocScore, includeCurve ? aucRocCurve : Collections.emptyList());
     }
 
-    private static double[] percentilesArray(Percentiles percentiles) {
+    private static double[] percentilesArray(Percentiles percentiles, String errorIfUndefined) {
```

Comment:
[my random thought]
So if IIUC in the case of all the label ending up having the same value (inlier or outlier), the "evaluate" request will fail.
Have you considered an alternative approach in which the error is returned as part of EvaluationMetricResult? This way, the request could succeed and other metrics (like precision, recall, etc.) could be calculated and returned to the user.

Reply:
It's a good one, I've thought about it too. However, it would take much more effort and I'm not convinced it's worth it as the workaround is simple and the chances for this edge case to happen are low. If we see users requesting a different behaviour here, we can make this change in the future.